src/uvbybeta.jl

```diff
@@ -0,0 +1,327 @@
+
+        if isnan(eby_in)
+            delm1 = m1zams - m1
+            delc1 = c1 - c1zams
+
+            if delm1 < 0
+                by0 = 2.946 - hbeta - 0.1 * delc1 - 0.25 * delm1
+            else
+                by0 = 2.946 - hbeta - 0.1 * delc1
+            end
+            eby_in = by - by0
+        end
+        by0, m0, c0, ub0 = deredd(eby_in, by, m1, c1, ub)
+        delm0 =  m1zams - m0
+        mv = mvzams - 9 * (c0 - c1zams)
+        te = 5040 / (0.771453 * by0 + 0.546544)
+    elseif n == 7
+        # For group 7 c1 is the luminosity indicator for a particular beta, while
+        # beta {or (b-y)0} indicates temperature.
+
+        # Where beta is not available iteration is necessary to evaluate a corrected
+        # (b-y) from which beta is then estimated.
+        if isnan(hbeta)
+            byinit = by
+            m1init = m1
+            for i in 1:1000
+                m1by = @evalpoly byinit 0.345 -1.32 2.5
+                bycorr = byinit + (m1by - m1init) / 2
+
+                if abs(bycorr - byinit) <= 0.0001
+                    break
+                end
+                byinit = bycorr
+                m1init = m1by
+            end
+            hbeta = @evalpoly byinit 2.96618 -1.32861 1.01425
+        end
+        # m1(ZAMS) and mv(ZAMS) are calculated according to Crawford (1975) with
+        # corrections suggested by Hilditch, Hill & Barnes (1983, MNRAS 204, 241)
+        # and Olson (1984, A&AS 57, 443).
+        m1zams = @evalpoly hbeta 46.4167 -34.4538 6.41701
+        mvzams = @evalpoly hbeta 324.482 -188.748 11.0494 5.48012
+        # c1(ZAMS) calculated according to Crawford (1975)
+        if hbeta <= 2.65
+            c1zams = 2 * hbeta - 4.91
+        else
+            c1zams = @evalpoly hbeta 72.879 -56.9164 11.1555
+        end
+
+        if isnan(eby_in)
+            dbeta = 2.72 - hbeta
+            eby_in = by - (0.222 - 0.05 * (c1 - c1zams) +
+                    (1.11 - (0.1 + 3.6 * (m1zams - m1))) * dbeta + 2.7 * dbeta^2)
+        end
+        by0, m0, c0, ub0 = deredd(eby_in, by, m1, c1, ub)
+        delm0 = m1zams - m0
+        mv = mvzams - (9 + 20 * dbeta) * (c0 - c1zams)
+        te = 5040/(0.771453 * by0 + 0.546544)
+    elseif n == 8
+        # Dereddening is done using color-color relations derived from
+        # Olson (1984, A&AS 57, 443)
+        if by <= 0.65
+            eby_in = (5.8651 * by - ub - 0.8975) / 4.3351
+        elseif 0.65 < by < 0.79
+            eby_in = (-0.7875 * by - c1 + 0.6585) / -0.9775
+            by0 = by - eby_in
+
+            if by0 < 0.65
+                eby_in = (5.8651 * by - ub - 0.8975) / 4.3351
+            end
+        else
+            eby_in = (0.5126 * by - c1 - 0.3645) / 0.3226
+            by0 = by - eby_in
+
+            if by0 < 0.79
+                eby_in = (-0.7875 * by - c1 + 0.6585) / -0.9775
+            end
+            by0 = by - eby_in
+
+            # This will never happen (unless both by and c1 are <0, but is
```

Comment:
The color index may be negative, I'm not sure what the discontinuity parameter is.  I'd be safe and keep this test.